### ISSUE TITLE
Repeated regex strings cause exception if "Only use first match" is not checked

### ISSUE THREAD

**Issue by wilsoc5** [jenkinsci/build-time-blame-plugin#10]
We have a build that runs a python script to execute parameterized tests. These tests output many of the same lines for each run of the test. i.e:
```
Creating files for run 0
Creating files for run 100
Creating files for run 200
.... [more output] ...
Creating files for run 0
Creating files for run 100
Creating files for run 200
.... [more output] ...
```
My expectation was to to parse the time duration for each of these parameterized tests by using the regex `.*Creating files for run 0.*` , label "Run Test" (following the `.*Started by.*` example). However, if the "Only use first match" is *not* checked for that regex, I get an exception (shown below). If "Only use first match is checked, there is no exception. I suppose I was expecting a "Run Test 0", "Run Test 1", ... "Run Test N" where N is the number of times that expression was seen to be parsed and placed on the graph. If this is not the intended use, please let me know, though it certainly seems useful. 

The Exception referenced above: 
```
javax.servlet.ServletException: org.apache.commons.jelly.JellyTagException: jar:file:/var/lib/jenkins/plugins/build-time-blame/WEB-INF/lib/build-time-blame.jar!/org/jenkins/ci/plugins/buildtimeblame/action/BlameAction/index.jelly:11:70: <st:include> No page found 'body.jelly' for class org.jenkins.ci.plugins.buildtimeblame.action.BlameAction
	at org.kohsuke.stapler.jelly.JellyClassTearOff.serveIndexJelly(JellyClassTearOff.java:117)
	at org.kohsuke.stapler.jelly.JellyFacet.handleIndexRequest(JellyFacet.java:127)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:735)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.MetaClass$10.dispatch(MetaClass.java:362)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.MetaClass$5.doDispatch(MetaClass.java:236)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:135)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:126)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:86)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
	at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.commons.jelly.JellyTagException: jar:file:/var/lib/jenkins/plugins/build-time-blame/WEB-INF/lib/build-time-blame.jar!/org/jenkins/ci/plugins/buildtimeblame/action/BlameAction/index.jelly:11:70: <st:include> No page found 'body.jelly' for class org.jenkins.ci.plugins.buildtimeblame.action.BlameAction
	at org.kohsuke.stapler.jelly.IncludeTag.doTag(IncludeTag.java:124)
	at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:269)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.apache.commons.jelly.tags.core.CoreTagLibrary$1.run(CoreTagLibrary.java:98)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.ReallyStaticTagLibrary$1.run(ReallyStaticTagLibrary.java:99)
	at org.kohsuke.stapler.jelly.CallTagLibScript$1.run(CallTagLibScript.java:99)
	at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
	at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:269)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.apache.commons.jelly.tags.core.CoreTagLibrary$1.run(CoreTagLibrary.java:98)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.apache.commons.jelly.tags.core.CoreTagLibrary$2.run(CoreTagLibrary.java:105)
	at org.kohsuke.stapler.jelly.CallTagLibScript.run(CallTagLibScript.java:120)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.CallTagLibScript$1.run(CallTagLibScript.java:99)
	at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
	at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:269)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.ReallyStaticTagLibrary$1.run(ReallyStaticTagLibrary.java:99)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.ReallyStaticTagLibrary$1.run(ReallyStaticTagLibrary.java:99)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.ReallyStaticTagLibrary$1.run(ReallyStaticTagLibrary.java:99)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.kohsuke.stapler.jelly.ReallyStaticTagLibrary$1.run(ReallyStaticTagLibrary.java:99)
	at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
	at org.apache.commons.jelly.tags.core.CoreTagLibrary$2.run(CoreTagLibrary.java:105)
	at org.kohsuke.stapler.jelly.CallTagLibScript.run(CallTagLibScript.java:120)
	at org.apache.commons.jelly.tags.core.CoreTagLibrary$2.run(CoreTagLibrary.java:105)
	at org.kohsuke.stapler.jelly.JellyViewScript.run(JellyViewScript.java:95)
	at org.kohsuke.stapler.jelly.DefaultScriptInvoker.invokeScript(DefaultScriptInvoker.java:63)
	at org.kohsuke.stapler.jelly.DefaultScriptInvoker.invokeScript(DefaultScriptInvoker.java:53)
	at org.kohsuke.stapler.jelly.JellyClassTearOff.serveIndexJelly(JellyClassTearOff.java:112)
	... 63 more
```

```
	org.jenkins-ci.main:jenkins-war:2.46.3
```

**Comment by wilsoc5** [comment]
Timestamper 1.8.8
Build Time Blame 1.1.0

**Comment by LTegtmeier** [comment]
I'll look into the exception when "Only use first match" is not checked. 

When that works as designed, it will not modify the label based on the number of times the regex was found. If the configured label for `*Creating files for run 0.*` is "Run Test 0" and "Creating files for run 0" occurs more than once, each result will be shown in the Last Successful Build Times table as "Run Test 0".

I believe the sum of the time taken for each match of `*Creating files for run 0.*` will be summed together in the Build Time Trend graph. I'll check on that as well.

**Comment by LTegtmeier** [comment]
I haven't been able to reproduce the issue yet. But, I have a couple more things to try. I'm not sure why this error would depend on the checkbox since it seems to be unable to find a file that would be loaded the same way regardless of the checkbox.

I did verify that the Build Time Trend graph and the Mean Successful Build Times table will show the sum of the time taken for the matched lines for each label.

If you want to simulate the effect you are looking for, you could add multiple entries with "Only use first match" checked and a different label on each one:

![split multiple matches](https://cloud.githubusercontent.com/assets/6852365/26786478/6f226a96-49cc-11e7-9de3-adb339ff7984.PNG)

**Comment by LTegtmeier** [comment]
Note: A line will only be assigned the first label that matches it and labels are processed from top to bottom.

**Comment by LTegtmeier** [comment]
I have been unable to reproduce this issue. Does it still occur? Do the other suggestions solve your problem?

**Comment by stella-lyra** [comment]
yes, it is

**Comment by LTegtmeier** [comment]
@stella-lyra, I'm not sure what you mean. If you're seeing this issue, can you explain what you are seeing and what you expected?

**Comment by LTegtmeier** [comment]
Reopening as it was found again in https://github.com/jenkinsci/build-time-blame-plugin/issues/15

**Comment by DKroot** [comment]
Moving discussion here from #15. I get an exception every time I check Only Use First Match = off:

```
org.apache.commons.jelly.JellyTagException: jar:file:/pardidata2/jenkins_home/plugins/build-time-blame/WEB-INF/lib/build-time-blame.jar!/org/jenkins/ci/plugins/buildtimeblame/action/BlameAction/index.jelly:11:70: <st:include> No page found 'body.jelly' for class org.jenkins.ci.plugins.buildtimeblame.action.BlameAction
	at org.kohsuke.stapler.jelly.IncludeTag.doTag(IncludeTag.java:124)
...
```

Jenkins and all plug-ins are up-to-date:
* Jenkins ver. 2.89.2
* Timestamper 1.8.8
* Build TIme Blame: 1.1.0

**Comment by LTegtmeier** [comment]
@DKroot, I'm still not able to recreate the issue with those versions. There's no good reason for Jenkins to be looking for a body.jelly for BlameAction, only BlameReport is rendered that way and the onlyUseFirstMatch flag shouldn't affect it anyway.

Can you provide a zip of the Jenkins build directory for a job with this behavior? It should be on the master under `${JENKINS_HOME}/jobs/${JOB_NAME}`.

**Comment by DKroot** [comment]
Sure. Here you go: [PIE-Prod-create.zip](https://github.com/jenkinsci/build-time-blame-plugin/files/1588028/PIE-Prod-create.zip)

**Comment by LTegtmeier** [comment]
Fixes were released in v1.1.2

**Comment by DKroot** [comment]
Great! It's working now. Thanks!
